public/javascripts/search.js

```diff
@@ -0,0 +1,105 @@
+var search = (function() {
+    var displayedSongs = m.prop(songs)
+    var playlistName = m.prop()
+    var inLibrary = m.prop(currentInLibrary)
+    var searchString = m.prop('')
+    var enter = function(elem, checkboxClass) {
+        $(elem).keyup(function(e) {
+            if (e.keyCode == 13) {
+                $("#search-button").click()
+            }
+        })
+    }
+
+    var searchBoxComponent = {
+        view: function(ctrl, args) {
+            return m('#searchInput.input-group', [
+                m('input.form-control[type=text]', {
+                    placeholder: 'Title, Lyric or Author',
+                    onchange: m.withAttr('value', searchString),
+                    config: function(elem, isInit, context) {
+                        if (!isInit) {
+                            enter(elem);
+                        }
+                    }
+                }),
+                m('span.input-group-btn', [
+                    m('button#search-button.btn.btn-success', {
+                        onclick: function() {
+                            window.location.href = '/search?q=' + searchString()
+                        }
+                    }, [
+                        m('i.glyphicon.glyphicon-search')
+                    ])
+                ])
+            ])
+        }
+    }
+
+
+    var searchResultComponent = {
+        view: function() {
+            if (displayedSongs().length === 0) {
+                return m('h4', {
+                        style: {'margin-top': '35px'}
+                    }, 'Songs not found')
```

Comment:
Create another simple component that can offer the user options when a song is not found. (not required to merge this code though).